src/main/java/com/sun/xml/messaging/saaj/util/JAXMStreamSource.java

```diff
@@ -14,18 +14,26 @@
 
 import javax.xml.transform.stream.StreamSource;
 
+import static com.sun.xml.messaging.saaj.soap.MessageImpl.SAAJ_MIME_SOAP_BODY_PART_SIZE_LIMIT;
+import static java.lang.String.format;
+
 
 /**
  *
  * @author Anil Vijendran
  */
 public class JAXMStreamSource extends StreamSource {
+
+    private static final Integer soapBodyPartSizeLimit;
```

Comment:
How can one set this on the per-message basis? Generally speaking avoid using statics for tools/frameworks supposed to be available through the server base classloader.

Reply:
Well, you can't.

I myself also prefer not to "reach out" to the environment for configuration and use statics for the values, but it seemed to be the pattern already in place, e.g. as with the `lazyContentLength`, also set in the static initializer.

I can change so it is possible to set this on a per-message basis. Do you prefer it as a constructor parameter? I.e. something like this:

```java
public JAXMStreamSource(InputStream is) throws IOException {
    this(is, null);
}

public JAXMStreamSource(InputStream is, Integer soapBodyPartSizeLimit) throws IOException {
    ..
}
```
And change the code which instantiates the `JAXMStreamSource` to resolve the size-value from `SAAJUtil.getSystemInteger(SAAJ_MIME_SOAP_BODY_PART_SIZE_LIMIT)`?

And there is a similar pattern in `MessageImpl`.

Reply:
> I myself also prefer not to "reach out" [..] but it seemed to be the pattern already in place

well, let's do it right for the new code and fix it for the old one (#120) - it does not have to be done as part of this PR, even though it would be nice ;-)

> Do you prefer it as a constructor parameter?

having additional constructor is fine with me. Do not think having getter/setter makes sense for this. Also note that the same should be done for the `Reader` variant.

> And change the code which instantiates the JAXMStreamSource...

Yes

Reply:
I will make these changes as soon as possible! 👍

Reply:
@lukasj I am a little bit unsure how to proceed this.
A JAXMStreamSource is instantiated from two places in SOAPPartImpl, which is the super class for SOAPPart1_1Impl and SOAPPart1_2Impl, each with two constructors, and their use can be further nested back to instances of things like MessageFactory, ElementFactory, SOAPFactory. To introduce `soapBodyPartSizeLimit` as a proper per-instance value seems like it will have quite a substantial impact, since the value essentially will have to be passed through many different classes and code paths from typically a factory and down to where it is used. The current design is based on environment-variables being pulled in where they are needed, and changing to a more pure form of passing arguments to objects I think requires a bit more thought out design. The case in point with the JAXMStreamSource is that changing its constructors to accept `soapBodyPartSizeLimit` as a parameter will require SOAPPart1_1Impl and SOAPPart1_2Impl to supply it. And they should not either resolve it from an environment variable. 

I am starting to think that such a change should not be part of this pull request. At least someone more familiar with the code base should probably govern such a change.

Reply:
@lukasj Or do you mean that a more appropriate place for the `saaj.mime.soapBodyPartSizeLimit` value to be resolved is by e.g. `SOAPPartImpl`, as it is a setting concerning a SOAP-specific issue (hence the name)? And then keeping the JAXMStreamSource more independent? That is absolutely a change relevant for this PR I am more than willing to do. I would just like some confirmation on this, or if you would like the `saaj.mime.soapBodyPartSizeLimit` value to be resolved at another place, before I make any further changes.

The value is as of now in this branch also resolved in [abstract MessageImpl](https://github.com/eclipse-ee4j/metro-saaj/pull/117/files#diff-77fc76ff7267a389bcb24e56e7000bb4R122). Which has v1.1 and v1.2 implementations, which are instantiated by MessageFactoryImpl, which is extended by SOAPMessageFactory1_1Impl and SOAPMessageFactory1_2Impl. Is it ok to keep resolving the `saaj.mime.soapBodyPartSizeLimit` value in MessageImpl, or should it be placed in these factories? If placed in the factory, there may also be appropriate to consider a way to configure these programmatically with say a setter? And then one must also figure out a strategy for precedence, if configuring through system properties should override the value, if set or not, in the factory. Or to just allow configuring using the system property. Decisions, decisions... 😝